src/lib/parser/core/PokemonSaveParser.ts

```diff
@@ -401,18 +550,34 @@ export class PokemonSaveParser {
 
   /**
    * Parse the complete save file and return structured data
+   * Now supports both file and memory input via WebSocket
    */
-  async parseSaveFile (input: File | ArrayBuffer | FileSystemFileHandle): Promise<SaveData> {
+  async parseSaveFile (input: File | ArrayBuffer | FileSystemFileHandle | MgbaWebSocketClient): Promise<SaveData> {
     await this.loadSaveFile(input)
 
+    // Memory mode: read directly from emulator memory
+    if (this.isMemoryMode && this.webSocketClient) {
+      const partyPokemon = await this.parsePartyPokemon()
+
+      return {
+        party_pokemon: partyPokemon,
+        player_name: 'MEMORY', // TODO: Read from memory if needed
+        play_time: { hours: 0, minutes: 0, seconds: 0 }, // TODO: Read from memory if needed
+        active_slot: 0, // Memory doesn't have multiple save slots
+        sector_map: new Map(), // Not applicable for memory parsing
+        rawSaveData: new Uint8Array(131072), // Standard GBA save size for compatibility
+      }
+    }
```

Comment:
create a todo for implementing memory support for the player name and playtime.
refactor our structure to support undefined sector_map and rawSaveData for the case of memory use of the parser